src/public/js/util.js

```diff
@@ -185,13 +264,88 @@ var UTIL = (function() {
         throw new Error('Division did not happen properly');
       }
     }
-    return remainder;
+
+    return remainder; // trimLeadingZeros breaks this -- length expectations
+  }
+
+  // Returns evaluation in GF(256) of polynomial at value
+  // polynomial is represented as a big-endian array, IE: [1,2,3] -> x^2 + 2x + 3
+  // value is represented as an integer
+  function polynomialEval(polynomial, value) {
+    var valuePower = 1; // updated to represent value^i during loop
+    var result = 0;
+    var term;
+    var coefficient;
+    for (var i = 1; i <= polynomial.length; i++) {
+      coefficient = polynomial[polynomial.length - i];
+      term = fieldMult(coefficient, valuePower);
+      result = result ^ term; // Galois extension field addition is xor with base field 2
+      valuePower = fieldMult(valuePower, value);
+    }
+    return result;
+  }
+
+  // Returns product of polynomial and scalar in GF(256)
+  // polynomial is represented as a big-endian array, IE: [1,2,3] -> x^2 + 2x + 3
+  // value is represented as an integer
+  function polynomialScale(polynomial, scalar) {
+    var scaledPolynomial = new Uint8Array(polynomial.length);
+    for (var i = 0; i < polynomial.length; i++) {
+      scaledPolynomial[i] = fieldMult(polynomial[i], scalar);
+    }
+    return trimLeadingZeros(scaledPolynomial);
```

Comment:
I think for this one you should trim on the parameter coming in, so var scaledPoly = trim(polynomial) then loop over scaledPoly multiplying each of its elements by scaler and replacing that element with the result.

If there are any leading 0s in result that weren't in the parameter other than if scaler is 0 something went horribly wrong.

Commenting this again as the other comment on it accidentally got squashed by a commit.